### PR TITLE
docs(http_client source): use auto-generated config docs

### PR DESCRIPTION
**Pull request by neuronull**
closes: https://github.com/vectordotdev/vector/issues/16053
epic: https://github.com/vectordotdev/vector/issues/14998

### CODE REVIEW

File: website/cue/reference/components/sources/http_client.cue [hunk 1]
```diff
@@ -51,84 +52,7 @@ components: sources: http_client: {
...
 
-	configuration: {
-		auth: configuration._http_auth & {_args: {
-			password_example: "${PASSWORD}"
-			username_example: "${USERNAME}"
-		}}
-		endpoint: {
-			description: "Endpoint to collect observability data from."
-			required:    true
-			warnings: ["You must explicitly add the path to your endpoint."]
```
Comment (neuronull) [inline website/cue/reference/components/sources/http_client.cue:62]: I asked about this offline. Flagging here to not forget. I don't think we have a way to add this with the configurable macros.

Comment (spencergilbert) [reply website/cue/reference/components/sources/http_client.cue:62]: Noting that @StephenWakely seemingly handled it like [so](https://github.com/vectordotdev/vector/pull/15901/files/56c1b3d2afd155a5fb8e5ff1ff33b0b489dbb1fc..900f277206b4a389808f0fd14f46189e0ea0e478#diff-667cf782642616aecc46a83c6630f7c4edb2a7a50a19108218ee30e609c47041R48-R60) - worth a doublecheck that this is the way to go though.

File: src/sources/http_client/client.rs [hunk 2]
```diff
@@ -38,23 +38,30 @@ use vector_core::{
 };
 
 /// Configuration for the `http_client` source.
+#[serde_as]
 #[configurable_component(source("http_client"))]
 #[derive(Clone, Debug)]
 pub struct HttpClientConfig {
-    /// Endpoint to collect events from. The full path must be specified.
-    /// Example: "http://127.0.0.1:9898/logs"
+    /// The HTTP endpoint to collect events from. The full path must be specified.
```
Comment (spencergilbert) [inline src/sources/http_client/client.rs:45]:
```suggestion
    /// The HTTP endpoint to collect events from.
    ///
    /// The full path must be specified.
```

File: website/cue/reference/components/sources/base/prometheus_scrape.cue [hunk 3]
```diff
@@ -102,10 +102,13 @@ base: components: sources: prometheus_scrape: configuration: {
 			type: array: items: type: string: {}
 		}
 	}
-	scrape_interval_secs: {
-		description: "The interval between scrapes, in seconds."
+	scrape_interval: {
```
Comment (spencergilbert) [inline website/cue/reference/components/sources/base/prometheus_scrape.cue:105]: Strikes me as a breaking change?

File: website/cue/reference/components/sources/base/http_client.cue [hunk 4]
```diff
@@ -176,14 +173,28 @@ base: components: sources: http_client: configuration: {
...
+				"X-My-Custom-Header": ["a", "vector", "of", "values"]
+			}]
+			options: "*": {
+				description: "An HTTP request header and it's value(s)."
+				required:    true
+				type: array: items: type: string: {}
+			}
+		}
+	}
+	interval: {
```
Comment (spencergilbert) [inline website/cue/reference/components/sources/base/http_client.cue:188]: Strikes me as a breaking change?

**Review by spencergilbert — CHANGES_REQUESTED** [review]
It looks like the user facing names of the options were changed. We should be able to keep the UX the same and leverage `serde` to trim off the `_sec` suffix from the internals.

**Review by spencergilbert — APPROVED** [review]

File: website/cue/reference/components/sources/http_client.cue (hunk 1 above)
Comment (neuronull) [reply website/cue/reference/components/sources/http_client.cue:62]: added here: https://github.com/vectordotdev/vector/pull/16054/commits/c0e1610e401b3e675ca6f0b63ed560de5fea6f44

Comment (StephenWakely) [reply website/cue/reference/components/sources/http_client.cue:62]: Yeah, that's the only way we can add warnings right now.

**Review by StephenWakely — APPROVED** [review]

**Pull request merged**